src/main/java/org/jenkinsci/plugins/gitclient/RemoteGitImpl.java

```diff
@@ -196,17 +197,17 @@ public void clearCredentials() {
 
     /** {@inheritDoc} */
     public void addCredentials(String url, StandardCredentials credentials) {
-        proxy.addCredentials(url, credentials); // credentials are Serializable
+        proxy.addCredentials(url, CredentialsProvider.snapshot(StandardCredentials.class, credentials)); // credentials are Serializable
```

Comment:
So i hve seen this work - but I'm at a bit of a loss as the snapshot provider would be the first that can handle `StandardCredentials` and the subclass may require a more specific type.
@stephenc is this correct use of this API?